docs/docs/security-in-gatsby.md

```diff
@@ -1,9 +1,95 @@
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Cross-Site Scripting (XSS)
+
+Cross-Site Scripting is a type of attack that injects a script or a unexpected link to another site into the client side of the application.
+
+JSX elements automatically escapes HTML tags by design. So, when you try to inject them as the example below:
+
+```js
+// highlight-next-line
+const username = `<script src='https://path/to/badness.js'></script>`
+
+const User = () => <p> hello {username}</p>
+```
+
+it will render `hello <script src='https://path/to/badness.js'></script>` as a string inside the p tag.
+
+In other hand, there could be some fields in your application that you will need to render the inner HTML tags, such as a content field in a blog or a comment in a comments section, that are built in rich-text editors.
+
+That's when you expose your application to XSS attacks, since the only way to render these HTML tags in React is by using the `dangerouslySetInnerHTML` prop:
+
+```js
+const CommentRenderer = comment => (
+  // highlight-next-line
+  <p dangerouslySetInnerHTML={{ __html: comment }} />
+) // dangerous indeed.
+```
+
+**How can it be prevented?**
+
+The most straightforward way to prevent the XSS attack is to sanitize the innerHTML string before dangerously setting it. Fortunately, there are npm packages that does the job, like [sanitize-html](https://www.npmjs.com/package/sanitize-html) and [DOMPurify](https://github.com/cure53/DOMPurify).
+
+## Cross-Site Request Forgery (CSRF)
+
+A web application that use cookies could be attacked by the CSRF exploit, which deceives a your user to execute actions by its name. By default, the browser "trusts" all the activity made validating the user's identity and therefore sending the associated cookies in every request.
+
+For example, assume that your users received an email with a malicious link:
+
+```html
+<a href="http://maliciouswebsite.xyz/">Get your coupon here</a>
+```
+
+Then, when the website opens, it could render this tag:
+
+```html
+<img src="http://yourecommerce.com/profile" />
+```
+
+which, instead of rendering an image, it sends the request with the authentication cookies and opens a security breach in their personal information.
+
+**How can it be prevented?**
+
+This could be prevented in three major ways:
+
+### Additional Session Token
+
+In addition to cookie validation, another approach could be added in the flow, such as an [OAuth2](https://oauth.net/2/).
+
+### Same-Site Cookies
+
+Configure the created cookie in a way that it could be only sent by the same domain. [Most moderns browsers support it with a backward compatibility](https://caniuse.com/#feat=same-site-cookie-attribute).
+
+### Anti-CSRF Tokens
+
```

Comment:
Maybe needs an example. not totally understanding how/why this works.

Reply:
Yes, totally. I add a example about how to configure it